ivy/func_wrapper.py

```diff
@@ -31,6 +31,7 @@
     "with_unsupported_dtypes",
     "handle_nans",
     "handle_mixed_function",
+    "auto_shift_device",
```

Comment:
let's rename this to `handle_device_shifting`, just for consistency

Reply:
Maybe lets rename it to `handle_device_shifting_for_arrays` to make it clear that it only changes device for arrays and not for the op being executed.